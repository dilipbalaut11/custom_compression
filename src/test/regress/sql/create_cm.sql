-- test storages
CREATE TABLE cmstoragetest(st1 TEXT, st2 INT);
ALTER TABLE cmstoragetest ALTER COLUMN st1 SET STORAGE EXTERNAL;
\d+ cmstoragetest
ALTER TABLE cmstoragetest ALTER COLUMN st1 SET STORAGE MAIN;
\d+ cmstoragetest
ALTER TABLE cmstoragetest ALTER COLUMN st1 SET STORAGE PLAIN;
\d+ cmstoragetest
ALTER TABLE cmstoragetest ALTER COLUMN st1 SET STORAGE EXTENDED;
\d+ cmstoragetest
DROP TABLE cmstoragetest;

CREATE TABLE cmdata(f1 text COMPRESSION pglz);
INSERT INTO cmdata VALUES(repeat('1234567890',1000));
INSERT INTO cmdata VALUES(repeat('1234567890',1001));

-- copy with table creation
SELECT * INTO cmmove1 FROM cmdata;

-- we update using datum from different table
CREATE TABLE cmmove2(f1 text COMPRESSION pglz);
INSERT INTO cmmove2 VALUES (repeat('1234567890',1004));
UPDATE cmmove2 SET f1 = cmdata.f1 FROM cmdata;

-- copy to existing table
CREATE TABLE cmmove3(f1 text COMPRESSION pglz);
INSERT INTO cmmove3 SELECT * FROM cmdata;

-- drop original compression information
DROP TABLE cmdata;

-- check data is okdd
SELECT length(f1) FROM cmmove1;
SELECT length(f1) FROM cmmove2;
SELECT length(f1) FROM cmmove3;

-- zlib compression
CREATE TABLE zlibtest(f1 TEXT COMPRESSION zlib);
CREATE TABLE zlibtest(f1 TEXT COMPRESSION zlib);
CREATE TABLE zlibtest(f1 TEXT COMPRESSION zlib);
INSERT INTO zlibtest VALUES(repeat('1234567890',1004));
INSERT INTO zlibtest VALUES(repeat('1234567890 one two three',1004));
SELECT length(f1) FROM zlibtest;

-- alter compression method with rewrite
ALTER TABLE cmmove2 ALTER COLUMN f1 SET COMPRESSION zlib;
\d+ cmmove2
ALTER TABLE cmmove2 ALTER COLUMN f1 SET COMPRESSION pglz;
\d+ cmmove2

DROP TABLE cmmove1, cmmove2, cmmove3, zlibtest;
