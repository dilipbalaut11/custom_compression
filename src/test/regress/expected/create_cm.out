-- test storages
CREATE TABLE cmstoragetest(st1 TEXT, st2 INT);
ALTER TABLE cmstoragetest ALTER COLUMN st1 SET STORAGE EXTERNAL;
\d+ cmstoragetest
                                      Table "public.cmstoragetest"
 Column |  Type   | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
--------+---------+-----------+----------+---------+----------+-------------+--------------+-------------
 st1    | text    |           |          |         | external | pglz        |              | 
 st2    | integer |           |          |         | plain    |             |              | 

ALTER TABLE cmstoragetest ALTER COLUMN st1 SET STORAGE MAIN;
\d+ cmstoragetest
                                      Table "public.cmstoragetest"
 Column |  Type   | Collation | Nullable | Default | Storage | Compression | Stats target | Description 
--------+---------+-----------+----------+---------+---------+-------------+--------------+-------------
 st1    | text    |           |          |         | main    | pglz        |              | 
 st2    | integer |           |          |         | plain   |             |              | 

ALTER TABLE cmstoragetest ALTER COLUMN st1 SET STORAGE PLAIN;
\d+ cmstoragetest
                                      Table "public.cmstoragetest"
 Column |  Type   | Collation | Nullable | Default | Storage | Compression | Stats target | Description 
--------+---------+-----------+----------+---------+---------+-------------+--------------+-------------
 st1    | text    |           |          |         | plain   | pglz        |              | 
 st2    | integer |           |          |         | plain   |             |              | 

ALTER TABLE cmstoragetest ALTER COLUMN st1 SET STORAGE EXTENDED;
\d+ cmstoragetest
                                      Table "public.cmstoragetest"
 Column |  Type   | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
--------+---------+-----------+----------+---------+----------+-------------+--------------+-------------
 st1    | text    |           |          |         | extended | pglz        |              | 
 st2    | integer |           |          |         | plain    |             |              | 

DROP TABLE cmstoragetest;
CREATE TABLE cmdata(f1 text COMPRESSION pglz);
INSERT INTO cmdata VALUES(repeat('1234567890',1000));
INSERT INTO cmdata VALUES(repeat('1234567890',1001));
-- copy with table creation
SELECT * INTO cmmove1 FROM cmdata;
-- we update using datum from different table
CREATE TABLE cmmove2(f1 text COMPRESSION pglz);
INSERT INTO cmmove2 VALUES (repeat('1234567890',1004));
UPDATE cmmove2 SET f1 = cmdata.f1 FROM cmdata;
-- copy to existing table
CREATE TABLE cmmove3(f1 text COMPRESSION pglz);
INSERT INTO cmmove3 SELECT * FROM cmdata;
-- drop original compression information
DROP TABLE cmdata;
-- check data is okdd
SELECT length(f1) FROM cmmove1;
 length 
--------
  10000
  10010
(2 rows)

SELECT length(f1) FROM cmmove2;
 length 
--------
  10000
(1 row)

SELECT length(f1) FROM cmmove3;
 length 
--------
  10000
  10010
(2 rows)

-- lz4 compression
CREATE TABLE lz4test(f1 TEXT COMPRESSION lz4);
INSERT INTO lz4test VALUES(repeat('1234567890',1004));
INSERT INTO lz4test VALUES(repeat('1234567890 one two three',1004));
SELECT length(f1) FROM lz4test;
 length 
--------
  10040
  24096
(2 rows)

-- alter compression method with rewrite
ALTER TABLE cmmove2 ALTER COLUMN f1 SET COMPRESSION lz4;
\d+ cmmove2
                                        Table "public.cmmove2"
 Column | Type | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
--------+------+-----------+----------+---------+----------+-------------+--------------+-------------
 f1     | text |           |          |         | extended | lz4         |              | 

ALTER TABLE cmmove2 ALTER COLUMN f1 SET COMPRESSION lz4;
\d+ cmmove2
                                        Table "public.cmmove2"
 Column | Type | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
--------+------+-----------+----------+---------+----------+-------------+--------------+-------------
 f1     | text |           |          |         | extended | lz4         |              | 

-- preserve old compression method
ALTER TABLE cmmove2 ALTER COLUMN f1 SET COMPRESSION pglz PRESERVE (lz4);
INSERT INTO cmmove2 VALUES (repeat('1234567890',1004));
\d+ cmmove2
                                        Table "public.cmmove2"
 Column | Type | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
--------+------+-----------+----------+---------+----------+-------------+--------------+-------------
 f1     | text |           |          |         | extended | pglz        |              | 

-- create compression method
CREATE COMPRESSION METHOD pglz2 HANDLER pglzhandler;
ALTER TABLE cmmove2 ALTER COLUMN f1 SET COMPRESSION pglz2;
INSERT INTO cmmove2 VALUES (repeat('1234567890',1004));
SELECT length(f1) FROM cmmove2;
 length 
--------
  10000
  10040
  10040
(3 rows)

\d+ cmmove2
                                        Table "public.cmmove2"
 Column | Type | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
--------+------+-----------+----------+---------+----------+-------------+--------------+-------------
 f1     | text |           |          |         | extended | pglz2       |              | 

DROP TABLE cmmove1, cmmove2, cmmove3, lz4test;
